src/components/app-field.js

```diff
@@ -1,64 +1,180 @@
+import { Table, Icon, Popup, Button, Input } from 'semantic-ui-react'
+import axios from 'axios'
+
+import { getCookie } from 'formula_one'
+
+import { urlAppHiddenDetail } from '../urls'
 
 import main from '../css/app-field.css'
 
 class AppField extends React.PureComponent {
+  constructor(props) {
+    super(props)
+    this.state = {
+      password_verified: this.props.password_verified,
+      formOpen: false,
+      password: '',
+      hiddenData: '',
+      passwordPlaceholder: 'Enter password',
+      passwordError: false,
+    }
+  }
+
+  handleSubmitError = (data) => {
+    this.setState({
+      password: '',
+      passwordPlaceholder: data,
+      passwordError: true,
+    })
+  }
+
+  handleOpen = () => this.setState({ formOpen: true })
+
+  handleClose = () =>
+    this.setState({
+      formOpen: false,
+      passwordError: false,
+      password: '',
+      passwordPlaceholder: 'Enter password',
+    })
+
+  handlePasswordChange = (e, { value }) => {
+    this.setState({
+      password: value,
+    })
+  }
+
+  handleSubmitSecret = () => {
+    if(this.state.password==''){
+      this.handleSubmitError("Enter something")
+      return
+    }
+
+    const headers = {
+      "X-CSRFToken": getCookie("csrftoken"),
+      "content-type": "application/json",
+    }
+
+    axios
+      .post(
+        urlAppHiddenDetail(),
+        {
+          password: this.state.password,
+          id: this.props.activeApp.data["id"],
+        },
+        { headers: headers }
+      )
+      .then((res) => {
+        this.setState({
+          password_verified: true,
+          hiddenData: res.data[this.props.field],
+        })
+      })
+      .catch((err) => this.handleSubmitError(err.response.data))
+  }
+
+  handleKeyDown = (event) => {
+    if(event.key === 'Enter'){
+      this.handleSubmitSecret() 
+    }
+  }
+
   handleClick = () => {
     let range = document.createRange()
     range.selectNodeContents(this.textArea)
     let sel = window.getSelection()
     sel.removeAllRanges()
     sel.addRange(range)
-    document.execCommand('copy')
+    document.execCommand("copy")
     sel.removeAllRanges()
   }
-  render () {
+  render() {
     const { activeApp, field, verboseName, editable } = this.props
     const { data } = activeApp
     return (
       <Table.Row>
         <Table.Cell>{verboseName}</Table.Cell>
+        {this.state.password_verified ? (
+          <Table.Cell>
+            <div styleName="desc-container">
+              <code ref={(textarea) => (this.textArea = textarea)}>
+                {this.state.hiddenData == ''
+                  ? data[field]
+                  : this.state.hiddenData}
+              </code>
+            </div>
+          </Table.Cell>
+        ) : (
+          <Table.Cell>
+            {this.state.formOpen ? (
+              <div className="appfield_div">
+                <Input
+                  styleName="appfield_input"
+                  type={"password"}
```

Comment:
When you enclose something in `{}`, you are essentially writing JS. So what's the rule for string? They should be in `''`.